halo2_proofs/src/dev.rs

```diff
@@ -912,6 +918,374 @@ impl<F: FieldExt> MockProver<F> {
         }
     }
 
+    /// Returns `Ok(())` if this `MockProver` is satisfied, or a list of errors indicating
+    /// the reasons that the circuit is not satisfied.
+    /// Constraints and lookup are checked at `usable_rows`, parallelly.
+    pub fn verify_par(&self) -> Result<(), Vec<VerifyFailure>> {
+        self.verify_at_rows_par(self.usable_rows.clone(), self.usable_rows.clone())
+    }
+
+    /// Returns `Ok(())` if this `MockProver` is satisfied, or a list of errors indicating
+    /// the reasons that the circuit is not satisfied.
+    /// Constraints are only checked at `gate_row_ids`,
+    /// and lookup inputs are only checked at `lookup_input_row_ids`, parallelly.
+    pub fn verify_at_rows_par<I: Clone + Iterator<Item = usize>>(
```

Comment:
Directly editing `iter()` -> `par_iter()` in `verify_at_rows` would make for a smaller diff. But I suppose that might cause conflicts in future rebases.

Reply:
Yeah, was trying to just add things and avoid editing things for easier rebasing.